monai/apps/pathology/transforms/post/array.py

```diff
@@ -0,0 +1,123 @@
+# You may obtain a copy of the License at
+#     http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Sequence, Union
+
+import numpy as np
+import torch
+
+from monai.config.type_definitions import NdarrayOrTensor
+from monai.data.meta_obj import get_track_meta
+from monai.networks.layers import GaussianFilter
+from monai.transforms.post.array import AsDiscrete, FillHoles, RemoveSmallObjects, SobelGradients
+from monai.transforms.transform import Transform
+from monai.utils import TransformBackends, convert_data_type, convert_to_tensor, optional_import
+from monai.utils.type_conversion import convert_to_dst_type
+
+label, _ = optional_import("scipy.ndimage.measurements", name="label")
+disk, _ = optional_import("skimage.morphology", name="disk")
+opening, _ = optional_import("skimage.morphology", name="opening")
+watershed, _ = optional_import("skimage.segmentation", name="watershed")
+
+__all__ = ["CalcualteInstanceSegmentationMap"]
+
+
+class CalcualteInstanceSegmentationMap(Transform):
+    """
+    Process Nuclei Prediction with XY Coordinate Map.
+
+    Args:
+        threshold_pred: threshold the float values of prediction to int 0 or 1 with specified theashold. Defaults to 0.5.
+        threshold_overall: threshold the float values of overall gradient map to int 0 or 1 with specified theashold.
+            Defaults to 0.4.
+        min_size: objects smaller than this size are removed. Defaults to 10.
+        sigma: std. could be a single value, or `spatial_dims` number of values. Defaults to 0.4.
+        kernel_size: the size of the Sobel kernel. Defaults to 17.
+        radius: the radius of the disk-shaped footprint. Defaults to 2.
+
+    """
+
+    backend = [TransformBackends.TORCH]
+
+    def __init__(
+        self,
+        threshold_pred: float = 0.5,
+        threshold_overall: float = 0.4,
+        min_size: int = 10,
+        sigma: Union[Sequence[float], float, Sequence[torch.Tensor], torch.Tensor] = 0.4,
+        kernel_size: int = 17,
+        radius: int = 2,
+    ) -> None:
+        self.sigma = sigma
+        self.radius = radius
+
+        self.sobel_gradient = SobelGradients(kernel_size=kernel_size)
+        self.remove_small_objects = RemoveSmallObjects(min_size=min_size)
+        self.pred_discreter = AsDiscrete(threshold=threshold_pred)
+        self.overall_discreter = AsDiscrete(threshold=threshold_overall)
+        self.fill_holes = FillHoles()
+
+    def __call__(self, prob_map: NdarrayOrTensor, hover_map: NdarrayOrTensor) -> NdarrayOrTensor:  # type: ignore
+        """
+        Args:
+            prob_map: the probability map output of the NP branch, shape must be [1, H, W, [D]].
```

Comment:
I'm not sure that we really need the optional depth channel. If someone is using it, the chances are that it is in error, so we might want to raise a ValueError and remove that D from the expected input

Reply:
Thanks for your advice! I have just added it in https://github.com/Project-MONAI/MONAI/pull/5173/commits/f506b0d1d9090d9268bbb7a7115d0ae5d3193eec.